libp2p/protocols/pubsub/gossipsub/behavior.nim

```diff
@@ -446,22 +450,24 @@ proc rebalanceMesh*(g: GossipSub, topic: string, metrics: ptr MeshMetrics = nil)
     let median = peers[medianIdx]
     if median.score < g.parameters.opportunisticGraftThreshold:
       trace "median score below opportunistic threshold", score = median.score
-      var avail = toSeq(
-        g.gossipsub.getOrDefault(topic, initHashSet[PubSubPeer]()) -
-        g.mesh.getOrDefault(topic, initHashSet[PubSubPeer]())
-      )
-
-      avail.keepIf do (x: PubSubPeer) -> bool:
-        # avoid negative score peers
-        x.score >= median.score and
-        # don't pick explicit peers
-        x.peerId notin g.parameters.directPeers and
-        # and avoid peers we are backing off
-        x.peerId notin g.backingOff.getOrDefault(topic)
-
-      # by spec, grab only 2
-      if avail.len > 2:
-        avail.setLen(2)
+      var avail: seq[PubSubPeer]
+      let currentMesh = g.mesh.getOrDefault(topic)
+
+      g.gossipSub.withValue(topic, peerList):
+        for it in peerList[]:
+          if
+            # avoid negative score peers
+            it.score >= median.score and
+            it notin currentMesh and
+            # don't pick explicit peers
+            it.peerId notin g.parameters.directPeers and
+            # and avoid peers we are backing off
+            it.peerId notin g.backingOff.getOrDefault(topic):
+              avail.add(it)
+
+              # by spec, grab only 2
+              if avail.len > 1:
```

Comment:
I do it the other way around :) conditions get 4 spaces, if block gets 2

Reply:
well, out guide says 2 space indent per block - since the block of the  if is the code and not the condition, it's more logical to move the condition an extra level - same goes for `proc` definitions - body = 2 spaces, parameters = 4 spaces - it's probably the best one can do with indent-based languages which also is why python auto-formatters tend to do it this way